api/v1alpha1/virtualmachine_conversion.go

```diff
@@ -879,13 +879,149 @@ func restore_v1alpha2_VirtualMachineReadinessProbeSpec(
 	}
 }
 
+func convert_v1alpha1_PreReqsReadyCondition_to_v1alpha2_Conditions(
+	dst *v1alpha2.VirtualMachine) []metav1.Condition {
+
+	var preReqCond, vmClassCond, vmImageCond, vmSetResourcePolicy, vmBootstrap *metav1.Condition
+	var preReqCondIdx int
+
+	for i := range dst.Status.Conditions {
+		c := &dst.Status.Conditions[i]
+
+		switch c.Type {
+		case string(VirtualMachinePrereqReadyCondition):
+			preReqCond = c
+			preReqCondIdx = i
+		case v1alpha2.VirtualMachineConditionClassReady:
+			vmClassCond = c
+		case v1alpha2.VirtualMachineConditionImageReady:
+			vmImageCond = c
+		case v1alpha2.VirtualMachineConditionVMSetResourcePolicyReady:
+			vmSetResourcePolicy = c
+		case v1alpha2.VirtualMachineConditionBootstrapReady:
+			vmBootstrap = c
+		}
+	}
+
+	// If we didn't find the v1a1 PrereqReady condition then there is nothing to do.
+	if preReqCond == nil {
+		return dst.Status.Conditions
+	}
+
+	// If any of the v1a2 conditions were already present, remove v1a1 PrereqReady condition and move on.
+	if vmClassCond != nil || vmImageCond != nil || vmSetResourcePolicy != nil || vmBootstrap != nil {
+		return append(dst.Status.Conditions[:preReqCondIdx], dst.Status.Conditions[preReqCondIdx+1:]...)
```

Comment:
Can we move this to L910 so we don't have to also repeat it at the end of the function too?

Reply:
done. double-check if it looks okay?.